flink-scala/src/main/scala/org/apache/flink/api/scala/codegen/TypeInformationGen.scala

```diff
@@ -238,10 +243,24 @@ private[flink] trait TypeInformationGen[C <: Context] {
       val elementTpe = $elementTypeInfo
       new TraversableTypeInfo($collectionClass, elementTpe) {
         def createSerializer(executionConfig: ExecutionConfig) = {
+          val elementSerializer = elementTpe.createSerializer(executionConfig)
+
+          // -------------------------------------------------------------------------------------
+          // NOTE:
+          // the following anonymous class is needed here, and should not be removed
+          // (although appears to be unused) since it is required for backwards compatibility
+          // with Flink versions pre 1.8, that were using Java deserialization.
+          // -------------------------------------------------------------------------------------
+          val unused = new TraversableSerializer[${desc.tpe}, ${desc.elem.tpe}](
+              elementSerializer,
+              $cbfStringLiteral) {
+
+                  override def legacyCbfCode = $cbfStringLiteral
+              }
+
           new TraversableSerializer[${desc.tpe}, ${desc.elem.tpe}](
-              elementTpe.createSerializer(executionConfig)) {
-            def getCbf = implicitly[CanBuildFrom[${desc.tpe}, ${desc.elem.tpe}, ${desc.tpe}]]
-          }
+                                  elementSerializer,
```

Comment:
I'm not sure if it'll matter at all, but maybe we should avoid sharing the created `elementSerializer` between the `unused` traversable serializer and the actual returned one.

Moreover, it _might_ (again, not 100% sure just a wild guess) have some unknown / undiscovered implications with the generated anonymous classname.

Reply:
One reason to do share it, is maybe the users would feel surprised if the serializer was created twice instead of once (i.e. log message, class path scanning etc', reflection and introspection)

But I don't know.. I'd go with your intuition here @tzulitai :-)